pkg/cmd/secvulnMaven.go

```diff
@@ -5,33 +5,299 @@
+		return pom, err
+	}
+
+	err = xml.Unmarshal(body, &pom)
+	if err != nil {
+		return pom, err
+	}
+
+	return pom, nil
+
+}
+
+func readPomFromRepo(repo, artifactName, version string) (Pom, error) {
+
+	var url string
+	if repo == "https://galasadev-cicsk8s.hursley.ibm.com/main/maven/obr" {
+		url = fmt.Sprintf("%s/dev/galasa/%s/%s/%s-%s.pom", repo, artifactName, version, artifactName, version)
+
+	} else if repo == "https://repo.maven.apache.org/maven2" {
+		// Future proofing for artifacts like com.jcraft.jsch - can be improved
+		s := strings.Split(artifactName, ".")
+		var artifact string
+		for i := 0; i < len(s); i++ {
+			artifact += s[i] + "/"
+		}
+		url = fmt.Sprintf("%s/%s%s/%s-%s.pom", repo, artifact, version, s[len(s)-1], version)
+	} else {
+		fmt.Println("Invalid repositories provided to galasabld")
+		panic(nil)
+	}
+
+	return readPomFromUrl(url)
+
+}
+
+func createPseudoMavenProject(pom Pom) {
+
+	createDirectory(pom.ArtifactId)
+
+	createPom(pom)
+
+	completedProjects = append(completedProjects, pom.ArtifactId)
+}
+
+func createDirectory(artifactName string) {
+
+	if err := os.Mkdir(fmt.Sprintf("%s/%s", secvulnMavenParentDir, artifactName), os.ModePerm); err != nil {
+		fmt.Printf("Unable to create directory for artifact %s - %v", artifactName, err)
+	}
+
+	// Change into the sub-directory
+	os.Chdir(fmt.Sprintf("%s/%s", secvulnMavenParentDir, artifactName))
+}
+
+func createPom(pom Pom) {
+	newPom := &Pom{}
+
+	newPom.GroupId = pom.GroupId
+	newPom.ArtifactId = pom.ArtifactId
+	newPom.Version = pom.Version
+	newPom.Packaging = "jar"
+
+	newPom.Parent = &Parent{
+		GroupId:    "dev.galasa",
+		ArtifactId: "security-scanning",
+		Version:    "0.21.0",
+	}
+
+	var dependencies []Dependency
+	for i := 0; i < len(pom.Dependencies.Dependencies); i++ {
+		groupId := pom.Dependencies.Dependencies[i].GroupId
+		if groupId == "dev.galasa" {
+			artifactId := pom.Dependencies.Dependencies[i].ArtifactId
+			version := pom.Dependencies.Dependencies[i].Version
+			dependency := &Dependency{
+				GroupId:    groupId,
+				ArtifactId: artifactId,
+				Version:    version,
+			}
+			dependencies = append(dependencies, *dependency)
+
+			// If a pseudo maven project hasn't been made for this dependency, add to the to do list
+			bool := checkIfCompleted(artifactId)
+			if bool == false {
+				toDoProjects = append(toDoProjects, artifactId)
+			}
+		}
+	}
+
+	if len(dependencies) > 0 {
+		newPom.Dependencies = &Dependencies{
+			Dependencies: dependencies,
+		}
+	}
+
+	filename := "pom.xml"
+	file, err := os.Create(filename)
+	if err != nil {
+		fmt.Printf("Unable to create pom.xml for artifact %s", pom.ArtifactId)
+		panic(err)
+	}
+
+	xmlWriter := io.Writer(file)
+
+	enc := xml.NewEncoder(xmlWriter)
+	enc.Indent("  ", "    ")
+	err = enc.Encode(newPom)
+	if err != nil {
+		fmt.Printf("Unable to encode the pom.xml for artifact %s", pom.ArtifactId)
+		panic(err)
+	}
+
+	// Change back to the parent directory
+	os.Chdir(secvulnMavenParentDir)
```

Comment:
dont use chdir in a program,  you don't want the change the working directory ever.
use the full path the filename instead.

Reply:
done